syntax = "proto2";

import "self_debug/proto/ast_parser.proto";
import "self_debug/proto/builder.proto";
import "self_debug/proto/dataset.proto";
import "self_debug/proto/llm_parser.proto";
import "self_debug/proto/llm_agent.proto";

package aws;


// NextId: 15
message Repo {
  optional string root_dir = 1;

  optional string github_url = 12;
  optional string base_commit_id = 13;

  optional string branch = 2 [default = "{source_branch}-{timestamp}"];
  // Old branch and clean up.
  optional string source_branch = 3 [default = "master"];
  optional bool git_clean = 4;
  optional bool git_restore = 5;

  // To find out Java base commit.
  optional int32 timeout_minutes = 6 [default = 90];
  optional int32 max_mvn_iterations = 7;
  optional bool run_java_metrics = 8;
  optional bool run_java_base_commit_search = 9;
  optional bool run_java_base_commit_search_no_maven = 10;
  optional bool run_java_hash = 11;
  optional bool run_repo_license = 14;
}


// NextId: 3
message FileSystemWriter {
  optional bool enable_feedback = 2 [default = true];
}


// NextId: 5
message TemplatePromptManager {
  optional string template_prompt = 1;
  optional string template_prompt_file = 2;
  optional string template_prompt_file_for_project = 4;

  // Used by `template_prompt_file`.
  repeated string required_fields = 3;
}

// NextId: 3
message PromptManager {
  oneof prompt_manager {
    TemplatePromptManager template_prompt_manager = 1;
  }
  optional int32 restart_messages_len_gt = 2;
}

// NextId: 13
message Config {
  optional LlmAgent llm_agent = 1;

  optional Dataset dataset = 10;
  optional Repo repo = 2;

  optional Builder builder = 3;
  optional AstParser ast_parser = 9;
  optional FileSystemWriter file_system_writer = 7;
  optional PromptManager prompt_manager = 4;
  optional LlmParserByGroup llm_parser_by_group = 5;

  optional int32 max_iterations = 6 [default = 50];
  // How many positive examples to add to the prompt.
  optional int32 max_n_examples = 12;
  optional int32 repeat = 8 [default = 1];
  optional bool max_migration = 11;
}
